app/src/main/java/org/simple/clinic/illustration/IllustrationSync.kt

```diff
@@ -2,21 +2,52 @@ package org.simple.clinic.illustration
 
 import io.reactivex.Completable
 import io.reactivex.Single
+import okhttp3.ResponseBody
+import org.simple.clinic.storage.files.FileOperations
 import org.simple.clinic.sync.ModelSync
 import org.simple.clinic.sync.SyncConfig
+import retrofit2.Call
+import retrofit2.Response
+import java.io.File
 import javax.inject.Inject
 import javax.inject.Named
 
 class IllustrationSync @Inject constructor(
     private val illustrations: List<HomescreenIllustration>,
-    @Named("sync_config_daily") private val configProvider: Single<SyncConfig>
+    private val fileDownloadService: FileDownloadService,
+    private val fileOperations: FileOperations,
```

Comment:
`IllustrationSync` should not be directly responsible for deciding which folder the images should be stored in. If we decide to change this, we'll have to change this in multiple places (the repository where the images are being read from, as well as this place) as well as makes testing harder because this class is doing too much (fetching the images as well as saving them).

We should delegate this task to the `HomescreenIllustrationRepository`. Could you please take a look at `ReportsSync`? We have done something similar there.

Reply:
Done